src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -282,14 +260,6 @@ HRESULT STDMETHODCALLTYPE CorProfiler::AssemblyLoadFinished(AssemblyID assembly_
     return S_OK;
   }
 
-  if (!is_attached_) {
-    return S_OK;
-  }
```

Comment:
Is it safe to remove this check?

Reply:
yeah, the check is still on line 298, after the lock guard, this was an optimisation to avoid the lock in case we already know that the profiler is not attached, but the common execution is the opposite.